week_3/project/week_3.py

```diff
@@ -82,21 +107,38 @@ def docker_config():
     },
 )
 
+
 docker_week_3_pipeline = week_3_pipeline.to_job(
     name="docker_week_3_pipeline",
     config=docker_config,
     resource_defs={
         "s3": s3_resource,
         "redis": redis_resource,
     },
+    op_retry_policy=RetryPolicy(max_retries=10, delay=1, backoff=None, jitter=None),
 )
 
 
-local_week_3_schedule = None  # Add your schedule
+local_week_3_schedule = ScheduleDefinition(job=local_week_3_pipeline, cron_schedule="*/15 * * * *")
+
 
-docker_week_3_schedule = None  # Add your schedule
+@schedule(cron_schedule="0 * * * *", job=docker_week_3_pipeline)
+def docker_week_3_schedule():
+    request = docker_week_3_pipeline.run_request_for_partition(partition_key="1", run_key=None)
+    yield request
 
 
-@sensor
-def docker_week_3_sensor():
-    pass
+@sensor(job=docker_week_3_pipeline, minimum_interval_seconds=30)
+def docker_week_3_sensor(context):
+    new_keys = get_s3_keys(bucket="dagster", prefix="prefix", endpoint_url="http://host.docker.internal:4566")
+    if len(new_keys) == 0:
```

Comment:
this appears to be unpythonic (though I personally prefer it...). I'll check to `if not new_keys`